attestation-agent/attestation-agent/src/lib.rs

```diff
@@ -68,44 +69,55 @@ pub trait AttestationAPIs {
 
 /// Attestation agent to provide attestation service.
 pub struct AttestationAgent {
-    config: Option<Config>,
+    _config: Config,
 }
 
 impl Default for AttestationAgent {
     fn default() -> Self {
-        let config = Config::try_from(config::DEFAULT_AA_CONFIG_PATH).ok();
-        AttestationAgent { config }
+        if let Ok(_config) = Config::try_from(config::DEFAULT_AA_CONFIG_PATH) {
+            return AttestationAgent { _config };
+        }
+
+        AttestationAgent {
+            _config: Config::default(),
+        }
     }
 }
 
 impl AttestationAgent {
     /// Create a new instance of [AttestationAgent].
-    pub fn new(config_path: &str) -> Self {
-        let config = Config::try_from(config_path).ok();
+    pub fn new(config_path: Option<&str>) -> Result<Self> {
+        let _config = match config_path {
```

Comment:
we don't need an underscore prefix, do we?

Reply:
We do need this. As if no features in `kbs` and `coco_as` is enabled, the config is not used thus the lint checker will raise a no-use warning.